pandas/tests/window/test_rolling.py

```diff
@@ -1950,3 +1950,38 @@ def test_numeric_only_corr_cov_series(kernel, use_arg, numeric_only, dtype):
         op2 = getattr(rolling2, kernel)
         expected = op2(*arg2, numeric_only=numeric_only)
         tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("unit", ["s", "ms", "us", "ns"])
+def test_rolling_rolling_sum_window_microseconds_confilict_timestamp(unit):
+    # GH#55106
+    df_time = DataFrame(
+        {"A": range(5)}, index=date_range("2013-01-01", freq="1s", periods=5)
```

Comment:
Could you also test where `date_range` has `tz="UTC"` using `pytest.mark.parameterize`?

Reply:
Surprisingly seems it has another issue by `tz="UTC"`. Should I make another issue and continue there? Meanwhile am investigating to see how it is happening

Reply:
If we can fix this within this PR that would be good

Reply:
I have added the patch considering the `UTC` for `_index_array` since it is returning the wrong values after casting `idx` for non-nanosecond values :) Have not found any other way at the moment to consider these values. I'm open to any other suggestions for handling the inner values in a more organised way.

Reply:
What you added now feels like a stop-gap solution, converting the non-nanoseconds datetimes to nanoseconds (which will only work as long as they are not out of bounds), while we should try to fix it properly by not having the convert the actual data, just as you did for tz-naive data. But I think we can fix this at the same place where you are construction the correct `win_freq_i8`, will comment there.